### ISSUE TITLE
Pass weights to loss function 

### ISSUE THREAD

**Issue by khanhptnk** [pytorch/pytorch#1319]
Currently loss functions does not allow weighting each instance differently. For example, in machine translation, we don't want to calculate losses for PAD labels. This can be done with cross entropy by specifying label weights, but can't be done with MSELoss (which is used to train critic/baseline model if we apply RL to MT). Manually writing a new loss function is possible, but having this feature is very convenient. The weights are better be per-instance rather than per-label.

**Comment by apaszke** [comment]
Duplicate of #264.
